src/vs/workbench/contrib/search/browser/search.contribution.ts

```diff
@@ -198,6 +198,11 @@ configurationRegistry.registerConfiguration({
 			description: nls.localize('search.quickOpen.includeSymbols', "Whether to include results from a global symbol search in the file results for Quick Open."),
 			default: false
 		},
+		'search.threads': {
+			type: 'number',
+			description: nls.localize('search.threads', "Number of threads to use for searching."),
```

Comment:
Could you add that `0` means that the engine will use automatically determine the number of threads to use? Also, I'm debating whether this is just something we should support for ripgrep, or whether we should pass this option to extension-contributed Text Search Providers also